sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs

```diff
@@ -976,6 +976,8 @@ public partial class BlobBaseClient
         public virtual System.Threading.Tasks.Task<Azure.Response<bool>> ExistsAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual Azure.Response<Azure.Storage.Blobs.Models.BlobProperties> GetProperties(Azure.Storage.Blobs.Models.BlobRequestConditions conditions = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response<Azure.Storage.Blobs.Models.BlobProperties>> GetPropertiesAsync(Azure.Storage.Blobs.Models.BlobRequestConditions conditions = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.IO.Stream OpenRead(long position = (long)0, int bufferSize = 1048576, Azure.Storage.Blobs.Models.BlobRequestConditions conditions = null) { throw null; }
+        public virtual System.Threading.Tasks.Task<System.IO.Stream> OpenReadAsync(long position = (long)0, int bufferSize = 1048576, Azure.Storage.Blobs.Models.BlobRequestConditions conditions = null) { throw null; }
```

Comment:
We don't actually need `OpenReadAsync()`, it has the same functionality of `OpenRead()`.  I only added it to follow our existing conventions.

Reply:
I wonder if it's actually in our benefit to have an `OpenReadAsync()` that does a `GetProperties()` call (or even a download at the given position argument). This would allow us to prepopulate the stream's `Length` asynchronously, as we can't asynchronously fetch a .NET property. By doing this, we obey the concept that seekability, `Position`, and `Length` come together.

Reply:
I would remove the `Async` overload as it doesn't add anything.

Reply:
I'm thinking we do the `GetProperties()` call to get the blob length either in `OpenRead()/OpenReadAsync()`, or in an initalization method called by `OpenRead()/OpenReadAsync()`, so we should keep the sync/async versions.

@tg-msft, do you think we should also fill the first buffer at this point?  If we went down this route, I'd create `LazyLoadingBlobStream.Initalize()`, which would be called from `OpenRead()/OpenReadAsync()`, and would download the first buffer full of data, and populate `LazyLoadingBlobStream.Length`.

Reply:
I think the extra `GetProperties` call is kind of a waste.  Could someone (eventually) call `Seek(0, ...)` if they wanted to force us to download the first bit of data, populate `Length`, etc.?

Reply:
@jaschrep-msft was concerned that a user may call `OpenRead()`, and then immediately call `.Length` on the returned stream.  Since we haven't fetched anything from the blob, Length will be reported as 0.

Reply:
I think that's fine.  It makes conceptual sense why a newly created read stream would have an unknown length.  I doubt most folks will care and the `GetProperties` call will be a waste.  I wouldn't mind a workaround like `Seek(0, ...)` to force an initial request if someone really needed it.